search/django/weedid/views.py

```diff
@@ -524,13 +545,29 @@ def awaiting_list(request):
     user = request.user
     if not (user and user.is_authenticated and user.is_staff):
         return HttpResponseForbidden("You dont have access to proceed")
-    awaiting_list = [
-        retrieve_listing_info(dataset, awaiting_review=True)
-        for dataset in Dataset.objects.filter(status="AR")
-    ]
-    return HttpResponse(json.dumps(awaiting_list))
+    try:
+        awaiting_list = [
+            retrieve_listing_info(dataset, awaiting_review=True)
+            for dataset in Dataset.objects.filter(status="AR")
+        ]
+        return HttpResponse(json.dumps(awaiting_list))
+    except Exception:
+        return HttpResponseNotAllowed("Awaiting list is not able to retrieve")
 
 
+def editing_list(request):
+    user = request.user
+    if not (user and user.is_authenticated):
+        return HttpResponseForbidden("You dont have access to proceed")
+    try:
+        editing_list = [
+            retrieve_listing_info(dataset, awaiting_review=False)
+            for dataset in Dataset.objects.filter(status="C", user=WeedidUser.objects.get(id=user.id))
```

Comment:
Is it reasonable to just return whether or not the user should be allowed to edit a dataset as a flag in the existing `upload_list` view?